### ISSUE TITLE
rmp destructors

### ISSUE THREAD

**Issue by Veix123** [UTNuclearRoboticsPublic/temoto2#1]
ensure that resource manager releases its resources properly when destroyed

**Comment by SouLeo** [comment]
What indication do you have that resources aren't being destroyed? I want to try and replicate your issue.

Also, in terms of editing source code, I should be looking within these files, correct?:

- temoto_2/include/rmp/resource_manager.h
- temoto_2/include/rmp/resource_client.h

**Comment by Veix123** [comment]
The issue is not related to pluginlib's ClassLoader.
The way resources are unloaded has been changed since I posted this issue, but I believe the problem is still relevant. The allocation of TeMoto resources is initiated by various interfaces (e.g. sensor_manager_interface, output_manager_interface, etc) and may pass multiple nodes until a specific external resource (launch file or linux process, etc) is allocated by the process_manager. When the process_manager detects a resource failure, a status message is being propagated back towards the interface that made the load call.
Now, when some relaying RMP node in the middle of this allocation chain is destroyed, the client connections called by this node will be automatically unloaded but all the incoming connections are being ignored (see the //todo in [resource_manager.h](https://github.com/UTNuclearRoboticsPublic/temoto2/blob/b0699a249757c427922f6635de0550f519c0a297/temoto_2/include/rmp/resource_manager.h#L199)). We should send a status message with a "resource closed" status message to all connections on the server side, but I remember experiencing a deadlock situation when trying to implement that.
I will close the issue and leave a mark under the To-Do project.
